plugins/out_s3/s3.c

```diff
@@ -1052,14 +1052,7 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
      * remove chunk from buffer list- needed for async http so that the
      * same chunk won't be sent more than once
      */
-    if (chunk) {
-        create_time = chunk->create_time;
-    }
-    else {
-        create_time = time(NULL);
-    }
-
-    ret = s3_put_object(ctx, tag, create_time, body, body_size);
+    ret = s3_put_object(ctx, tag, chunk->first_log_time, body, body_size);
```

Comment:
So the code above was checking if we actually have a `chunk` or if it's null. I think we still need an equivalent check here. Am I wrong? Will we always have the chunk? I expect you to read and understand the existing code.

Reply:
So @Claych read through the code and it indeed looks like there's no way to get into this function without having a chunk that's non-null. However, to make this explicit Clay and I agreed he will add a check for the chunk being non-null here: https://github.com/fluent/fluent-bit/blob/v1.9.10/plugins/out_s3/s3.c#L2174